arangod/RestServer/VocbaseContext.cpp

```diff
@@ -50,66 +51,68 @@ VocbaseContext::~VocbaseContext() {
 VocbaseContext* VocbaseContext::create(GeneralRequest& req, TRI_vocbase_t& vocbase) {
   // _vocbase has already been refcounted for us
   TRI_ASSERT(!vocbase.isDangling());
-
+  
+  // superusers will have an empty username. This MUST be invalid
+  // for users authenticating with name / password
+  bool isSuperUser = req.authenticated() && req.user().empty() &&
+                     req.authenticationMethod() == rest::AuthenticationMethod::JWT;
+  if (isSuperUser) {
+    return new VocbaseContext(req, vocbase, ExecContext::Type::Internal,
+                              /*sysLevel*/ auth::Level::RW,
+                              /*dbLevel*/ auth::Level::RW);
+  }
+  
   AuthenticationFeature* auth = AuthenticationFeature::instance();
   TRI_ASSERT(auth != nullptr);
-  if (auth == nullptr) {
-    return nullptr;
-  } else if (!auth->isActive()) {
-    return new VocbaseContext(req, vocbase, /*flags*/ 0,
+  if (!auth->isActive()) {
+    if (ServerState::readOnly() && !isSuperUser) {
```

Comment:
AFAIK the second part of the condition `!isSuperUser` is always true here, and should be removed for clarity.